dashboard/app/models/user_level.rb

```diff
@@ -97,6 +98,11 @@ def handle_unsubmit
     end
   end
 
+  # clear the per-request cache when user levels are saved or updated
+  def clear_user_level_cache
+    user.clear_user_level_cache unless user.nil
+  end
```

Comment:
There are tests that cause user_levels to get loaded and cached, then directly insert new user_levels and expect to see different results. This pointed out a theoretical problem: if in the course of a request, a user_level got created, and then a different caller asked for user_levels again before the end of the request, the new user_level would not be returned. Using the ActiveRecord callback to clear the cache on user_level insert or update is simple and robust.